### PR TITLE
Add event system

### PR DESCRIPTION
**Pull request by fcheval**

### CODE REVIEW

File: officemix/officemix.py [hunk 1]
```diff
@@ -105,6 +109,16 @@ def studio_submit(self, data, suffic=''):
         
         return {'result': 'success'}
 
+    @XBlock.json_handler
+    def emit_event(self, data, suffix=''):
```
Comment (tusbar) [inline officemix/officemix.py]: why not use https://github.com/edx/xblock-utils/blob/master/xblockutils/publish_event.py?

Comment (fcheval) [inline officemix/officemix.py]: Done

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 requests
 
 -e .
+-e git+ssh@github.com:edx/xblock-utils.git#egg=xblock-utils
```
Comment (tusbar) [inline requirements.txt]:
```
+e git+https://github.com/edx/xblock-utils.git#egg=xblock-utils
```

File: officemix/officemix.py [hunk 3]
```diff
@@ -91,11 +91,15 @@ def student_view(self, context=None):
             embed_code=embed_code,
             display_name=cgi.escape(display_name)))
 
+        js_str = self.resource_string("/static/js/officemix.js")
```
Comment (tusbar) [inline officemix/officemix.py]: why a variable here and not below?

File: officemix/static/js/officemix.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  player.on('ready', function () {
+      console.log('loaded');
+
+      player.getDuration(function (duration) {
+          console.log('duration:', duration);
+          var handlerUrl = runtime.handlerUrl(element, 'publish_event');
+          var data = {
+              'event_type': 'xblock.officemix.loaded',
+              'video': iframe.get(0).src,
```
Comment (tusbar) [inline officemix/static/js/officemix.js:11]: prefer iframe.attr('src'), but use a variable out of this scope so it’s not computed a thousand times

File: officemix/static/js/officemix.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+function OfficeMixBlock(runtime, element) {
+  var iframe = $('iframe', element);
+  var player = new playerjs.Player(iframe.get(0));
+
+  player.on('ready', function () {
+      console.log('loaded');
+
+      player.getDuration(function (duration) {
+          console.log('duration:', duration);
```
Comment (tusbar) [inline officemix/static/js/officemix.js]: remove the debug.

File: officemix/static/js/officemix.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+          var handlerUrl = runtime.handlerUrl(element, 'publish_event');
```
Comment (tusbar) [inline officemix/static/js/officemix.js:8]: this does not have to be computed on each event.

**Comment by tusbar** [comment]
This has to be finished by 3pm today. Can you please fix the remaining comments?

**Pull request closed without merging**